app/services/retrieve_record_service.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+# We can remove this service (and just call search_service.fetch directly)
+# after we migrate to Blacklight 8
+class RetrieveRecordService
+  # :reek:DuplicateMethodCall
+  # :reek:FeatureEnvy
+  def retrieve(search_service, ...)
+    if using_blacklight7?
+      _response, document = search_service.fetch(...)
+      document
+    else
+      search_service.fetch(...)
+    end
+  end
+
+  private
+
+    # :reek:UtilityFunction
+    def using_blacklight7?
+      Gem.loaded_specs['blacklight'].version.to_s.start_with? '7'
```

Comment:
This doesn't need to be blocking, but would it make sense to memoize this method? Or maybe even use an initializer to set this, so we don't have to check every time?